nvflare/private/fed/server/server_engine.py

```diff
@@ -215,6 +215,22 @@ def start_app_on_server(self, run_number: str, job_id: str = None, job_clients=N
             self.engine_info.status = MachineStatus.STARTED
             return ""
 
+    def _recv(self, conn, job_id, run_number):
```

Comment:
I think the main purpose of job_id and run_number arguments are for the log msg.  The underlying recv function does not need job_id and run_number.
Maybe using a few try/except inside _listen_command is better.  It's just my personal opinion.

Reply:
that's correct, So you prefer not wrap this into a try/except clause, but how do you distinguishing different errors as the main _listen_command has several _conn.recv_ conn.send and conn.aux_send methods
for example, conn.aux_send also do the similar try/expcet/raise to capture local error message.  

If might hurt the performance bit, but comparing to the training/transfer data, this should be nothing, I thought it would give a much better context of the error. 

That being said, I am willing to cancel PR if you all felt this is not necessary @YuanTingHsieh @yhwen @nvidianz

Reply:
I am fine with this PR.

I think the server engine code needs to be cleaned up in 2.2 so maybe these will be changed at that time.

Reply:
If don't make this PR, we have to do another PR to capture the exception cause, currently the exception is swallowed. did not show the real cause of the exception